resources/patches/silicon-edits.patch

```diff
@@ -0,0 +1,117 @@
+ .../common}/collections/immutable/MultiMap.scala   |  0
+ .../immutable/MyInsertionOrderedSet.scala.bak      |  0
+ .../common}/collections/immutable/package.scala    |  0
+ .../common}/collections/mutable/MultiMap.scala     |  0
+ .../main/scala/common}/collections/package.scala   |  0
+ .../main/scala/common}/concurrency/package.scala   |  0
+ .../main/scala/common}/config/package.scala        |  0
+ .../main/scala/common}/io/package.scala            |  0
+ .../scala => src/main/scala/common}/package.scala  |  0
+ 10 files changed, 8 insertions(+), 6 deletions(-)
+ rename {common/src/main/scala => src/main/scala/common}/collections/immutable/MultiMap.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/collections/immutable/MyInsertionOrderedSet.scala.bak (100%)
+ rename {common/src/main/scala => src/main/scala/common}/collections/immutable/package.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/collections/mutable/MultiMap.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/collections/package.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/concurrency/package.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/config/package.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/io/package.scala (100%)
+ rename {common/src/main/scala => src/main/scala/common}/package.scala (100%)
+
+diff --git a/build.sbt b/build.sbt
+index 8ba98c80..6f213f1c 100644
+--- a/build.sbt
++++ b/build.sbt
+@@ -4,23 +4,24 @@
+ //
+ // Copyright (c) 2011-2019 ETH Zurich.
+ 
++resolvers += MavenCache("local-maven", file(Path.userHome.absolutePath + "/.m2/repository"))
++
+ // Import general settings from Silver
+ lazy val silver = project in file("silver")
+ 
+-lazy val common = (project in file("common"))
+-  .dependsOn(silver)
+-
+ // Silicon specific project settings
+ lazy val silicon = (project in file("."))
+   .dependsOn(silver % "compile->compile;test->test")
+-  .dependsOn(common)
+-  .aggregate(common)
+   .settings(
+     // General settings
+     name := "Silicon",
+     organization := "viper",
+     version := "1.1-SNAPSHOT",
+ 
++    // Publish settings
++    publishArtifact := true,
++    publishTo := Some(MavenCache("local-maven", file(Path.userHome.absolutePath + "/.m2/repository"))),
++
+     // Compilation settings
+     // Remove elidable method calls such as in SymbExLogger during compilation
+     // scalacOptions ++= Seq("-Xelide-below", "1000"),
+@@ -30,7 +31,7 @@ lazy val silicon = (project in file("."))
+     libraryDependencies += "com.typesafe.scala-logging" %% "scala-logging" % "3.9.2",
+     libraryDependencies += "org.apache.commons" % "commons-pool2" % "2.9.0",
+     libraryDependencies += "io.spray" %%  "spray-json" % "1.3.6",
+-    libraryDependencies += "com.microsoft.z3" % "z3" % "4.8.7" from "https://www.sosy-lab.org/ivy/org.sosy_lab/javasmt-solver-z3/com.microsoft.z3-4.8.7.jar",
```

Comment:
Why is the hard-coding of the Z3 version and origin necessary?

Reply:
I think it is something internal to `silicon` the hard-coded version number. 

Regarding the origin, it seems that `z3` is shipped with the [JAR file](http://leodemoura.github.io/blog/2012/12/10/z3-for-java.html), but maybe the SBT tool is not able to resolve the location of the file (it should depend on the used package manager)

Reply:
More info on the hard-coded version number: https://github.com/viperproject/silicon/commit/b192d39ae3bdc58d02633bc8de350bce7aec6bfe